[docs] adds new FAQ entries (#1752)
carbon-language/carbon-lang: docs/project/faq.md

design. If we had written prototype code with `.car` instead of `.carbon`, the
core language design would still proceed. What does matter is that we _pick_ one
of the options, and try to do so with some rationale and consistency.

It can be tempting to revisit these, especially when there's community
disagreement. However, because the language is still very incomplete, it is
important that we focus our time and energy on the open questions. This
is one reason why we're often a bit reluctant to revisit some of the more minor
decisions. It's not that they're _definitely_ right, or that they are truly set
in stone. It's just a prioritization of making progress on the open questions
because at least we have an answer and that answer is probably good enough for
now. The signal to really revisit stuff is new information. There was an example
with digit separators: we had never considered how many _domain_ conventions we
would want to support and so had overly restricted where the separators go.

We encourage folks to either shrug and let quirky syntax be quirky, or look for
the objective use case that was clearly missed in the previous decision and

> makes the current state _sharply_ problematic before pushing for change.

Reviewer: I feel like we should say something along the lines of:

When there is disagreement between options, comments saying "I like X" or "I don't like X" don't really contribute to the conversation. Much more helpful would be a reason for or against X that hasn't already been considered.

Author: Done.

    interoperability wrappers.
-   Test-driven verification that migrations are correct.

> ### Where should I ask questions about Carbon Language?

Reviewer 2: Would this make sense under "What is Carbon's status?", right after the demo question?

Please ask questions and hold discussions either by using
[GitHub Discussions](https://github.com/carbon-language/carbon-lang/discussions)

> or the
> [Carbon Language Discord server](https://discord.com/channels/655572317891461132/996113200464543804).

Reviewer 2: ```suggestion
or
[#language-questions on Discord](https://discord.com/channels/655572317891461132/998959756045713438).
```

I admit I'm not sure whether we should point at the server versus the server invite, but let's use #language-questions.

GitHub Issues should be reserved for missing features, bugs, and anything else
that is fixable by way of a Pull Request.

> ### What are the prerequisites for contributing to Carbon Language's design and tools??

Reviewer 2: ```suggestion
### What are the prerequisites for contributing to Carbon Language's design and tools?
```

to appeal to everyone's preferences. We try to base the decision on consistent
rationale, which can be found in the proposals which are linked throughout the
design.

We understand that many decisions won't be universally popular. Different
developers weigh factors differently. However, we need to commit to a choice in
order to make progress on the language design as a whole. This doesn't mean that
they're _definitely_ right or set in stone, it just means we need to focus our
time and energy on other issues.

Sometimes, it will be appropriate to revisit a decision; for example, when there
is new information introduced, a new community joins Carbon, or there is an
order of magnitude growth in the community. We will do this by filing an issue
for leads with what new information is available, and we will reevaluate the
choice.

For example, we're doing this with digit separators: we missed important
_domain_ conventions and had overly restricted where separators are allowed,
[an issue was filed with the new information](https://github.com/carbon-language/carbon-lang/issues/1485),
and we're fixing the choice.

> When there is disagreement between options, comments saying "I like X" and "I

Reviewer 2: ```suggestion
When there is a discussion about options or disagreement with a decision, comments saying "I like X" and "I
```

WDYT of this? "disagreement between options" is feeling odd to me.

> When there is a discussion about options or disagreement with a decision, comments saying "I like X" and "I
> don't like X" do not contribute to the conversation. We encourage folks to
> either shrug and let quirky syntax be quirky, or look for the objective use case
> that was clearly missed in the previous decision and makes the current state
> _sharply_ problematic before pushing for change.

Reviewer 3: Maybe also recommend using reactions to capture sentiment? I'm thinking something like:

```suggestion
When there is a discussion about options or disagreement with a decision, comments saying "I like X" and "I
don't like X" do not contribute to the conversation. You should use the reactions to signal this kind of sentiment, and we'll pay attention to those. Beyond that, we encourage folks to
either shrug and let quirky syntax be quirky, or look for the objective use case
that was clearly missed in the previous decision and makes the current state
_sharply_ problematic before pushing for change.
```
